node/src/ui_gateway/websocket_supervisor.rs

```diff
@@ -388,23 +418,79 @@ impl WebSocketSupervisorReal {
         ok::<(), ()>(())
     }
 
-    fn send_to_clients(clients: Vec<(u64, &mut dyn ClientWrapper)>, json: String) {
-        clients.into_iter().for_each(|(client_id, client)| {
-            match client.send(OwnedMessage::Text(json.clone())) {
-                Ok(_) => match client.flush() {
-                    Ok(_) => (),
-                    Err(_) => warning!(
-                        Logger::new("WebSocketSupervisor"),
-                        "Client {} dropped its connection before it could be flushed",
-                        client_id
-                    ),
-                },
-                Err(e) => error!(
+    fn send_to_clients(
+        clients: Vec<(u64, &mut dyn ClientWrapper)>,
+        json: String,
+    ) -> Option<Vec<SendToClientWebsocketError>> {
+        let errors: Vec<SendToClientWebsocketError> = clients
+            .into_iter()
+            .flat_map(|(client_id, client)| {
+                match client.send(OwnedMessage::Text(json.clone())) {
+                    Ok(_) => match client.flush() {
+                        Ok(_) => None,
+                        Err(e) => Some(SendToClientWebsocketError::FlushError((client_id, e))),
+                    },
+                    Err(e) => {
+                        error!(
+                            Logger::new("WebSocketSupervisor"),
+                            "Error sending to client {}: {:?}", client_id, e
+                        );
+                        //TODO do we want to treat this error more in detail or to return None would be acceptable?
+                        Some(SendToClientWebsocketError::SendError((client_id, e)))
```

Comment:
If you can explore the client.send() code and conclude that SendError can't happen in real life in this context, then don't worry too much about handling it.  If it _could_ happen, but just _doesn't_ very much, then it should at least be logged.

Reply:
It's hard to find a clear answer. I have nothing with exact error examples because there are traits with so many impls and I got lost in there. However I found what I think could be sufficient for now even if it's quite general: 

    _If the sink encounters an error other than being temporarily full, it
    uses the `Err` variant to signal that error. In most cases, such errors
    mean that the sink will permanently be unable to receive items._
    
The second sentence leads me to a thought that it may be allowed for me to simply discard that client the way I did with the flush error.